sdk/eventhub/azure-eventhub-checkpointstoretable/tests/test_storage_table_partition_manager.py

```diff
@@ -1,8 +1,148 @@
+            ownership['owner_id'] = 'owner_id'
+            ownership['partition_id'] = str(i)
+            ownership['etag'] = None
+            ownership['last_modified_time'] = None
+            ownership_list.append(ownership)
+    newownershiplist = checkpoint_store.claim_ownership(ownership_list)
+    assert len(newownershiplist) == len(ownership_list)
+    for i in range(len(ownership_list)):
+        assert ownership_list[i]['etag'] != newownershiplist[i]['etag']
+        assert ownership_list[i]['last_modified_time'] != newownershiplist[i]['last_modified_time']
+
+    ownership_list = checkpoint_store.list_ownership(fully_qualified_namespace,eventhub_name,consumer_group)
+    assert len(ownership_list) == ownership_cnt
+    assert len(ownership_list) == len(newownershiplist)
+    for i in range(len(newownershiplist)):
+        assert ownership_list[i]['last_modified_time'] == newownershiplist[i]['last_modified_time']
+
+    ownership = [{'fully_qualified_namespace': 'test_namespace', 'eventhub_name': 'eventhub', 'consumer_group': '$default',
+     'owner_id': 'Bill', 'partition_id': '0', 'etag': newownershiplist[0]['etag'], 'last_modified_time': newownershiplist[0]['last_modified_time']}]
+    ownership_list = checkpoint_store.claim_ownership(ownership)
+    assert ownership_list[0]['owner_id'] == 'Bill'
+
+    ownership = [{'fully_qualified_namespace': 'test_namespace', 'eventhub_name': 'eventhub', 'consumer_group': '$default',
+     'owner_id': 'Jack', 'partition_id': '0', 'etag': 'W/"datetime\'2021-08-02T00%3A46%3A51.7645424Z\'"', 'last_modified_time': newownershiplist[0]['last_modified_time']}]
+    with pytest.raises(OwnershipLostError) as e_info:
+            checkpoint_store.claim_ownership(ownership)
+
+    ownership = [{'fully_qualified_namespace': 'test_namespace', 'eventhub_name': 'eventhub', 'consumer_group': '$default',
+     'owner_id': 'Jack', 'partition_id': '10', 'etag': 'W/"datetime\'2021-08-02T00%3A46%3A51.7645424Z\'"', 'last_modified_time': newownershiplist[0]['last_modified_time']}]
+    with pytest.raises(OwnershipLostError) as e_info:
+            checkpoint_store.claim_ownership(ownership)
+
+def _update_and_list_checkpoint(storage_connection_str, table_name):
+    fully_qualified_namespace = 'test_namespace'
+    eventhub_name = 'eventhub'
+    consumer_group = '$default'
+    partition_cnt = 8
+
+    checkpoint_store = TableCheckpointStore.from_connection_string(
+        storage_connection_str, table_name)
+    checkpoint_list = checkpoint_store.list_checkpoints(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(checkpoint_list) == 0
+    for i in range(partition_cnt):
+            checkpoint = {
+                'fully_qualified_namespace': fully_qualified_namespace,
+                'eventhub_name': eventhub_name,
+                'consumer_group': consumer_group,
+                'partition_id': str(i),
+                'offset': 2,
```

Comment:
Since the documentation states that `offset` is a string, can you add a test to ensure that update_checkpoint handles a string offset correctly?

Reply:
will do that